### PR TITLE
[#2] Add dockerfile

### PR DESCRIPTION
**Pull request by arael82**
closes https://github.com/entropy-code/entropay-employees/issues/2

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,2 +1,12 @@
+# Build
+FROM maven:3.8.4-openjdk-17 AS build-env
+WORKDIR /app
+COPY pom.xml ./
+RUN mvn dependency:go-offline
+COPY . ./
+RUN mvn package
```
Comment (umarcostuc) [inline Dockerfile:7]: I think is better to use build scope and do it outside docker file, and create the docker image with a maven docker plugin.
Here in Docker file I'd just copy the resulting .jar.

File: pom.xml [hunk 2]
```diff
@@ -11,7 +11,7 @@
 	<groupId>com.entropy</groupId>
 	<artifactId>entropay.employees</artifactId>
```
Comment (umarcostuc) [inline pom.xml:12]: entropay looks more like part of the groupId? I'd use `-` instead of `.`

Comment (reconeazaretzky) [reply pom.xml:12]: on top of what @umarcostuc said, we should use the domain of the company. Therefore, please use `com.entropyteam.entropay` as the `groupId`

File: pom.xml [hunk 3]
```diff
@@ -33,13 +32,27 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
```
Comment (reconeazaretzky) [inline pom.xml:36]: please do not use lombok

File: pom.xml [hunk 4]
```diff
@@ -33,13 +32,27 @@
...
+			<artifactId>lombok</artifactId>
+			<version>1.18.22</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-lang3</artifactId>
```
Comment (reconeazaretzky) [inline pom.xml:43]: are you already using it? I'll prefer to postpone until we find the need adding more libraries

File: Dockerfile (hunk 1 above)
Comment (agustinb4) [reply Dockerfile:7]: Good point, Including the build in the docker file, we make deploying easier, just needing a vm with docker. I think it not has impact from development perspective. 
Do you think we should change anyway??

Comment (umarcostuc) [reply Dockerfile:7]: the FROM openjdk:17 is deleting previous one?
The final image should be the smallest and cleanest possible without build deps if this is accomplished we could have it here.
Other thing to have in mind, with maven you could trigger the whole release process, as deploy the artifact, create the docker image, push it, increase the version, etc.
I'm not sure this is the right place to do it, please let me know if I'm missing something

Comment (arael82) [reply Dockerfile:7]: I am not very convinced to have the docker image creation attached to maven build until we discuss where to push it. Perhaps a spike is required to clarify these definitions together?

Comment (agustinb4) [reply Dockerfile:7]: yeah it creates 2 different images one with the build and one only with the jar. We only need to update the pom version in each release as part of the release script. 
I think for the moment we can start with this approach.

**Review by agustinb4 — APPROVED** [review]

**Pull request merged**